src/components/base/b-virtual-scroll/props.ts

```diff
@@ -0,0 +1,241 @@
+ */
+
+import type iItems from 'components/traits/i-items/i-items';
+import type { CreateFromItemFn } from 'components/traits/i-items/i-items';
+
+import type bVirtualScroll from 'components/base/b-virtual-scroll/b-virtual-scroll';
+
+import type {
+
+	VirtualScrollState,
+	ComponentDb,
+	RequestQueryFn,
+	ShouldPerform,
+	ComponentItemFactory,
+	ComponentItemType,
+	$ComponentRefs
+
+} from 'components/base/b-virtual-scroll/interface';
+
+import { defaultShouldProps, componentItemType } from 'components/base/b-virtual-scroll/const';
+
+import { ComponentTypedEmitter, componentTypedEmitter } from 'components/base/b-virtual-scroll/modules/emitter';
+import { ComponentInternalState } from 'components/base/b-virtual-scroll/modules/state';
+import { SlotsStateController } from 'components/base/b-virtual-scroll/modules/slots';
+import { ComponentFactory } from 'components/base/b-virtual-scroll/modules/factory';
+import { Observer } from 'components/base/b-virtual-scroll/modules/observer';
+
+import iData, { component, prop, system } from 'components/super/i-data/i-data';
+
+/**
+ * A class that is friendly to {@link bVirtualScroll}.
+ * It contains the properties of the {@link bVirtualScroll} component.
+ */
+@component()
+export default abstract class iVirtualScrollProps extends iData {
+	/** {@link iItems.item} */
+	readonly Item!: object;
+
+	/** {@link iItems.Items} */
+	readonly Items!: Array<this['Item']>;
+
+	/** {@link iItems.item} */
+	@prop({type: [String, Function]})
+	readonly item?: iItems['item'];
+
+	/** {@link iItems.itemKey} */
+	@prop({type: [String, Function]})
+	readonly itemKey?: CreateFromItemFn<object, string>;
+
+	/** {@link ComponentItemType} */
+	@prop({type: [String, Function]})
+	readonly itemType: keyof ComponentItemType | CreateFromItemFn<object, ComponentItemType> = componentItemType.item;
+
+	/** {@link iItems.itemProps} */
+	@prop({type: [Function, Object], default: () => ({})})
+	readonly itemProps!: iItems['itemProps'];
+
+	/**
+	 * Specifies the number of times the `tombstone` component will be rendered.
+	 *
+	 * This prop can be useful if you want to render multiple `tombstone` components
+	 * using a single specified element. For example, if you set `tombstonesSize` to 3,
+	 * then three `tombstone` components will be rendered on your page.
+	 */
+	@prop(Number)
+	readonly tombstonesSize?: number;
+
+	/**
+	 * This factory function is used to pass information about the components that need to be rendered.
+	 * The function should return an array of arbitrary length consisting of objects that satisfy the
+	 * `ComponentItem` interface.
+	 *
+	 * By default, the rendering strategy is based on the `chunkSize` and `iItems` trait.
+	 * In other words, the default implementation takes a data slice of length `chunkSize`
+	 * and calls the `iItems` functions to generate a `ComponentItem` object.
+	 *
+	 * However, nothing prevents the client from implementing any strategy by overriding this function.
```

Comment:
Исходная формулировка для английского языка нестандартная, можно переписать так: 
```suggestion
	 * However, the client can implement any required strategy by overriding this function.
```